Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUAdapterImpl.cpp

```diff
@@ -224,18 +231,23 @@ void AdapterImpl::requestDevice(const DeviceDescriptor& descriptor, CompletionHa
         SET_VALUE_32(minUniformBufferOffsetAlignment)
         SET_VALUE_32(minStorageBufferOffsetAlignment)
         SET_VALUE_32(maxVertexBuffers)
-        SET_VALUE_32(maxBufferSize)
+        SET_VALUE_64(maxBufferSize)
         SET_VALUE_32(maxVertexAttributes)
         SET_VALUE_32(maxVertexBufferArrayStride)
         SET_VALUE_32(maxInterStageShaderComponents)
         SET_VALUE_32(maxInterStageShaderVariables)
         SET_VALUE_32(maxColorAttachments)
+        SET_VALUE_32(maxColorAttachmentBytesPerSample)
         SET_VALUE_32(maxComputeWorkgroupStorageSize)
         SET_VALUE_32(maxComputeInvocationsPerWorkgroup)
         SET_VALUE_32(maxComputeWorkgroupSizeX)
         SET_VALUE_32(maxComputeWorkgroupSizeY)
         SET_VALUE_32(maxComputeWorkgroupSizeZ)
         SET_VALUE_32(maxComputeWorkgroupsPerDimension)
+        else {
+            requestInvalidDevice();
+            return;
+        }
 
 #undef SET_VALUE_32
 #undef SET_VALUE_64
```

Comment:
There has got to be a better way of doing this without the preprocessor.

Reply:
We could use templates to avoid have the _32 and _64 variants. Not immediately obvious how to get rid of the string conversion without some reflection information